client/lib/signup/step-actions/index.js

```diff
@@ -355,9 +362,11 @@ export function addPlanToCart( callback, dependencies, stepProvidedItems, reduxS
 		return;
 	}
 
+	const state = reduxStore.getState();
+	const cartItems = get( getSignupDependencyStore( state ), 'cartItems', null ) || [];
 	const providedDependencies = { cartItem };
 
-	const newCartItems = [ cartItem ].filter( ( item ) => item );
+	const newCartItems = [ cartItem ].filter( ( item ) => item ).concat( cartItems );
```

Comment:
We discussed this topic on Slack. My arguments are that when we make multiple simultaneous API calls to the server, we can have an out-of-order `update_blog_option` that would break the signup flow. That's why the original approach to use individual API request functions didn't work.

A potential refactor here would submit the cart items at the end of the flow, and not at the end of each step.